db/post_migrate/20210726130556_set_user_stats_draft_count.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class SetUserStatsDraftCount < ActiveRecord::Migration[6.1]
+  def change
```

Comment:
Sorry quick comment so we don’t need to wait another day of back and fourth.

I think this needs to be a migration and not a post migration. Otherwise, the counts  will appear broken until the deploy is completed and post migrations have run. Also, we’ve been bitten multiple times by a single giant update query where the query ends up timing out on large site. I think it’ll be better to migrate in batches instead.

Reply:
I made this a post-migration because I was concerned that it could be a bit more slower than I expected on gigantic sites. Making this run in batches is quite difficult because of the subquery. The LIMIT and OFFSET will make this query run very slow.

Reply:
> I made this a post-migration because I was concerned that it could be a bit more slower than I expected on gigantic sites.

Since the query might be slow, it is more important that we run it in a migration instead. Migrations are run before the new instances are deployed so it is better to blow up early than to blow up after we deploy the new code and the post migrations fail. Everyone will be shown a draft count of 0 until we can fix the post migration.



> Making this run in batches is quite difficult because of the subquery. The LIMIT and OFFSET will make this query run very slow.

Looks like migrations on user_stats should be quite fast based on a [previous query](https://github.com/discourse/discourse/blob/main/db/migrate/20191220134101_add_distinct_badge_count_to_user_stats.rb) that we run. Let's keep the query as it is but move it to a migration?